Update screenshots and fix old wording from OData  example (#61)
mulesoft/docs-apikit: modules/ROOT/pages/creating-an-odata-api-with-apikit.adoc

endif::[]
:keywords: apikit, apikit extension, odata

The APIkit OData Extension supports OData (Open Data Protocol) to create HTTP-based data services.

> The APIkit OData Extension builds the services from the entity data model (EDM), which OData services use to formalize their resource descriptions. You can write the EDM in RAML using EDM-compatible data types to represent API entities.

Reviewer: APIkit OData extension? (I'm not sure)

> The APIkit OData extension generates Mule flows to handle OData requests. You can use the APIkit OData Extension to, for example, expose a legacy API as an OData API, orchestrate data before exposing it in Salesforce, or create a bidirectional OData API for Oracle, DB2, or Azure DB.

Reviewer: Here it is APIkit OData extension (I think this is right)

> The APIkit OData extension generates Mule flows to handle OData requests. You can use the APIkit OData Extension to, for example, expose a legacy API as an OData API, orchestrate data before exposing it in Salesforce, or create a bidirectional OData API for Oracle, DB2, or Azure DB.

Reviewer: ```suggestion
The APIkit OData extension generates Mule flows to manage OData requests. You can use the APIkit OData extension to, for example, expose a legacy API as an OData API, orchestrate data before exposing it in Salesforce, or create a bidirectional OData API for Oracle, DB2, or Azure DB.
```

The OData MySQL example on this page shows how to use an HTTP-based data service to query a MySQL database.

== Prerequisites

> The following software is required to create and using an OData REST API with APIkit:

Reviewer: ```suggestion
The following software is required to create and use an OData REST API with APIkit:
```

* OData Plugin
* Mule EE 4.1.1 and later
* Anypoint Studio 7.1.4 and later

Additionally, you must download the link:{attachmentsdir}/odata.raml[OData example entity data model].

== Install the APIKit OData Extension

> To install the APIKit OData Extension in Anypoint Studio:

Reviewer: ```suggestion
To install the APIKit OData extension in Anypoint Studio:
```

. Select *Help* > *Install New Software*.

> . Select *Add...*

Reviewer: ```suggestion
. Click *Add...*
```

+
image::add-update-site.png[]
. In the *Name* field, type `APIkit for ODATA Update Site`.
. In the *Location* field, type `+http://studio.mulesoft.org/s4/apikit-for-odata/+`.

> . Select *Add*.

Reviewer: ```suggestion
. Click *Add*.
```

+
image::add-repository-odata.png[]

> . Select the *Anypoint ODATA Plugin* box, and select *Next*.

Reviewer: ```suggestion
. Select the *Anypoint ODATA Plugin* box, and click *Next*.
```

+
image::select-odata-extension.png[]

> . Accept the terms and conditions of the product, then select *Finish*.

Reviewer: ```suggestion
. Accept the terms and conditions of the product, then click *Finish*.
```

. Restart Studio to complete the installation.

== Implement an API Using the APIkit OData Extension

> The following example shows you how to implement an OData enabled REST API:

Reviewer: ```suggestion
The following example shows you how to implement an OData-enabled REST API:
```

. In Studio, select *File* > *New* > *Mule Project*.
. In the *Project Name* field, type the name of your Mule project.

> . Select *Finish*.

Reviewer: ```suggestion
. Click *Finish*.
```

. Copy the `odata.raml` file you downloaded in the prerequisites section to the `/src/main/resources/api` directory in your project.
. In the project explorer, right-click `odata.raml` and select *Mule* > *Generate OData API from RAML Types*.
+
image::generate-odata.png[]

> . The OData Extension generates the necessary files for youe OData API imlementation:

Reviewer: ```suggestion
. The OData extension generates the necessary files for your OData API implementation:
```

+
image::list-of-odata-files.png[]
+

> <1> The Mule configuration file with the flows for the API implementation.

Reviewer: ```suggestion
<1> The Mule configuration file with the flows for the API implementation
```

> <1> The generated RAML definition.

Reviewer: ```suggestion
<1> The generated RAML definition
```

> <1> The generated RAML libraries.

Reviewer: ```suggestion
<1> The generated RAML libraries
```

. Add the necessary logic for your API implementation in the flows.

== OData MySQL Example

The OData MySQL example is a fully functional OData API packaged as a Maven project. In this example, the data source, the app, and the service are local.

First, you install a MySQL database and load tables using a provided script. Next, you import the compressed project into Studio, which includes a MySQL database driver. You can examine the flows to see how to implement the endpoints you need by accessing the data. Then you run the project and call the REST and OData services.

=== Install and Populate a MySQL Database

. Install a MySQL database and launch MySQL.

> . Download MySQL project example link:{attachmentsdir}/apikit-odata-example-master.jar[apikit-odata-example-master.jar].

Reviewer: ```suggestion
. Download the MySQL project example link:{attachmentsdir}/apikit-odata-example-master.jar[apikit-odata-example-master.jar].
```

. Run the `example.sql` script located in `src/main/resources` using MySQL commands to load data into a database named `apikit-odata-example`.

=== Implement the OData API

. In Studio, Select *File* > *Import*.
. Select *Anypoint Studio* > *Packaged mule application (.jar)* > *Next*.

> . Browse for the `apikit-odata-example-master.jar` file and select *Finish*.

Reviewer: ```suggestion
. Browse for the `apikit-odata-example-master.jar` file and click *Finish*.
```
